pygmt/tests/test_solar.py

```diff
@@ -0,0 +1,131 @@
+"""
+Tests for solar.
+"""
+import datetime
+
+import pytest
+from pygmt import Figure
+from pygmt.exceptions import GMTInvalidInput
+
+
+@pytest.mark.mpl_image_compare
+def test_solar_default_terminator():
+    """
+    Test passing the solar argument with a time string and no terminator type
+    to confirm the default terminator type.
+    """
+    fig = Figure()
+    fig.solar(
+        region="d",
+        projection="W0/15c",
+        frame="a",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    return fig
+
+
+@pytest.mark.mpl_image_compare
+def test_solar_terminators():
+    """
+    Test passing the solar argument with a time string and no terminator type
+    to confirm the default terminator type.
+    """
+    fig = Figure()
+    fig.solar(
+        region="d",
+        projection="W0/15c",
+        frame="a",
+        terminator="d",
+        pen="1/blue",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    fig.solar(
+        region="d",
+        projection="W0/15c",
+        frame="a",
+        terminator="a",
+        pen="1/red",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    fig.solar(
+        region="d",
+        projection="W0/15c",
+        frame="a",
+        terminator="c",
+        pen="1/green",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
+    fig.solar(
+        region="d",
+        projection="W0/15c",
+        frame="a",
+        terminator="n",
+        pen="1/yellow",
+        terminator_datetime="1990-02-17 04:25:00",
+    )
```

Comment:
1. Calling `Figure.solar()` with `frame="a"` draws the frame and annotations. The old code draw the frame and annotations four times. Calling `Figure.basemap` separately is better.
2. pen attributes should be comma-separated.
```suggestion
    fig = Figure()
    fig.basemap(region="d", projection="W0/15c", frame="a")
    fig.solar(
        terminator="d",
        pen="1p,blue",
        terminator_datetime="1990-02-17 04:25:00",
    )
    fig.solar(
        terminator="a",
        pen="1p,red",
        terminator_datetime="1990-02-17 04:25:00",
    )
    fig.solar(
        terminator="c",
        pen="1p/green",
        terminator_datetime="1990-02-17 04:25:00",
    )
    fig.solar(
        terminator="n",
        pen="1p,yellow",
        terminator_datetime="1990-02-17 04:25:00",
    )
```